libraries/Parsers/Microsoft.Bot.Builder.Parsers.LU/Parser/LUResource.cs

```diff
@@ -0,0 +1,407 @@
+
+        [JsonProperty("contextText")]
+        public string ContextText { get; set; }
+
+        [JsonProperty("range")]
+        public Range Range { get; set; }
+
+        [JsonProperty("references", NullValueHandling = NullValueHandling.Ignore)]
+        public Reference References { get; set; }
+    }
+
+    public partial class Reference
+    {
+        [JsonProperty("source")]
+        public string Source { get; set; }
+    }
+
+    public partial class EntityElement
+    {
+        [JsonProperty("type")]
+        [JsonConverter(typeof(StringEnumConverter))]
+        public TypeEnum Type { get; set; }
+
+        [JsonProperty("entity")]
+        public string Entity { get; set; }
+
+        [JsonProperty("role")]
+        public string Role { get; set; }
+
+        [JsonProperty("startPos", NullValueHandling = NullValueHandling.Ignore)]
+        public int? StartPos { get; set; }
+
+        [JsonProperty("endPos", NullValueHandling = NullValueHandling.Ignore)]
+        public int? EndPos { get; set; }
+    }
+
+    internal static class Converter
+    {
+        public static readonly JsonSerializerSettings Settings = new JsonSerializerSettings
+        {
+            MetadataPropertyHandling = MetadataPropertyHandling.Ignore,
+            DateParseHandling = DateParseHandling.None,
+            Converters =
+            {
+                SectionTypeConverter.Singleton,
+                TypeEnumConverter.Singleton,
+                new IsoDateTimeConverter { DateTimeStyles = DateTimeStyles.AssumeUniversal }
+            },
+        };
+    }
+
+    internal class SectionTypeConverter : JsonConverter
+    {
+        public static readonly SectionTypeConverter Singleton = new SectionTypeConverter();
+
+        public override bool CanConvert(Type t) => t == typeof(SectionType) || t == typeof(SectionType?);
+
+        public override object ReadJson(JsonReader reader, Type t, object existingValue, JsonSerializer serializer)
+        {
+            if (reader.TokenType == JsonToken.Null)
+            {
+                return null;
+            }
+
+            var value = serializer.Deserialize<string>(reader);
+            switch (value)
+            {
+                case "entitySection":
+                    return SectionType.EntitySection;
+                case "newEntitySection":
+                    return SectionType.NewEntitySection;
+                case "simpleIntentSection":
+                    return SectionType.SimpleIntentSection;
+                case "nestedIntentSection":
+                    return SectionType.NestedIntentSection;
+                case "importSection":
+                    return SectionType.ImportSection;
+                case "modelInfoSection":
+                    return SectionType.ModelInfoSection;
+                case "qnaSection":
+                    return SectionType.QnaSection;
+            }
+
+            throw new Exception("Cannot unmarshal type SectionType");
```

Comment:
strongly consider adding the type name into the exception type. That way the error is more actionable.